src/parsing/filterVisitor.js

```diff
@@ -0,0 +1,92 @@
+import { FIELDS_MAP } from 'parsing/filterParser';
+import { getOriginalString } from 'parsing/parsingHelpers';
+
+function defaultValueOperatorFor(value) {
+  return (operator, fieldValue) => {
+    switch (operator) {
+      case ':':
+      case '=':
+        return value === fieldValue;
+      case '<':
+        return fieldValue < value;
+      case '<=':
+        return fieldValue <= value;
+      case '>':
+        return fieldValue > value;
+      case '>=':
+        return fieldValue >= value;
+      default:
+        throw new Error(`Unrecognized operator ${operator}`);
+    }
+  };
+}
+
+export function getFilterVisitorFromParser(parser) {
+  const BaseCstVisitor = parser.getBaseCstVisitorConstructorWithDefaults();
+  class FilterVisitor extends BaseCstVisitor {
+    constructor() {
+      super();
+      this.validateVisitor();
+    }
+
+    filter(ctx) {
+      const { children } = ctx;
+      const filters = [];
+      for (const ruleName of Object.keys(children)) {
+        if (ruleName !== 'EOF') {
+          const rules = children[ruleName];
+          filters.push(
+            ...rules.map((rule) => {
+              const negation = rule.children.$negation[0].children;
+              const negated = Object.keys(negation).length !== 0;
+              const fieldAbbrv = getOriginalString(rule.children.$field[0].children);
+              const operator = getOriginalString(rule.children.$operator[0].children);
+              const vChildren = rule.children.$value[0].children;
+              const value = this.visit(Object.values(vChildren)[0]);
+              const field = FIELDS_MAP[fieldAbbrv];
+              if (!field) {
+                throw new Error(`No matching field for abbreviation ${fieldAbbrv}`);
+              }
+              if (!value) {
+                throw new Error(`ValueType for ${field} not implemented yet`);
+              }
+              // TODO: Correctly lookup field
+              return (card) => {
+                const fieldValue = field.split('.').reduce((obj, fieldName) => obj[fieldName], card);
+                const result = value(operator, fieldValue);
+                if (negated) {
+                  return !result;
+                }
+                return result;
+              };
+            }),
+          );
+        }
+      }
+      return (card) => filters.every((filter) => filter(card));
+    }
+
+    // eslint-disable-next-line class-methods-use-this
```

Comment:
why are we ignoring the eslint here?

Reply:
Because the visit methods need to be class methods to be found correctly, but the class is stateless and they don't recurse so they never use `this` which is an eslint error.